src/RendezvousTracker.ts

```diff
@@ -0,0 +1,290 @@
+                        fileName = fileNameAsXml;
+                    }
+                }
+            }
+
+            this.clientPathsMap[fileName] = {
+                clientPath: this.convertDebuggerPathToClient(fileName),
+                clientLines: {
+                    [lineNumber]: this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber))
+                }
+            };
+        } else if (!this.clientPathsMap[fileName].clientLines[lineNumber]) {
+            // Add new client line to clint path map
+            this.clientPathsMap[fileName].clientLines[lineNumber] = this.convertDebuggerLineToClientLine(fileName, parseInt(lineNumber));
+        }
+
+        return fileName;
+    }
+
+    /**
+     * Helper function used to create a new RendezvousHistory object with default values
+     */
+    private createNewRendezvousHistory(): RendezvousHistory {
+        return {
+            hitCount: 0,
+            occurrences: {},
+            totalTime: 0.00,
+            type: 'historyInfo',
+            zeroCostHitCount: 0
+        };
+    }
+
+    /**
+     * Helper function to assist in the creation of a RendezvousLineInfo
+     * @param fileName processed file name
+     * @param lineNumber occurrence line number
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private createLineObject(fileName: string, lineNumber: number, duration?: string): RendezvousLineInfo {
+        return {
+            clientLineNumber: lineNumber,
+            clientPath: this.clientPathsMap[fileName].clientPath,
+            hitCount: 1,
+            totalTime: this.getTime(duration),
+            type: 'lineInfo'
+        };
+    }
+
+    /**
+     * Helper function to convert the duration to a float or return 0.00
+     * @param duration how long the rendezvous took to complete, if not supplied it is assumed to be zero
+     */
+    private getTime(duration?: string): number {
+        return duration ? parseFloat(duration) : 0.000;
+    }
+}
+
+export interface RendezvousHistory {
+    hitCount: number;
+    occurrences: {[key: string]: RendezvousFileInfo};
+    totalTime: number;
+    type: ElementType;
+    zeroCostHitCount: number;
+}
+
+interface RendezvousFileInfo {
+    hitCount: number;
+    occurrences: {[key: string]: RendezvousLineInfo};
+    totalTime: number;
+    type: ElementType;
+    zeroCostHitCount: number;
+}
+
+interface RendezvousLineInfo {
+    clientLineNumber: number;
+    clientPath: string;
+    hitCount: number;
+    totalTime: number;
+    type: ElementType;
+}
+
+interface RendezvousBlocks {
+    [key: string]: {
+        fileName: string;
+        lineNumber: string;
+    };
+}
+
+type ElementType = 'historyInfo' | 'fileInfo' | 'lineInfo';
+
+interface RendezvousClientPathMap {
+    [key: string]: RendezvousClientFile;
+}
+
+interface RendezvousClientFile {
+    [key: string]: string | RendezvousClientLineMap;
```

Comment:
This is still a bit muddy as far as typing is concerned. Do yo ureally need to make this a map that holds two separate things? Is there a way to break this out so that you have a property for whatever holds the [key:string]:string, and separate property for the [key:string]:number? I'm open to hearing a justifcation for this, but I try very hard to restrict properties to represent a single type. And it's not like `string | number`, this is `dictionary of string` OR `dictionary of dictionary of number`, which makes it a bit more confusing.

Reply:
This interface has been combined into the RendezvousClientPathMap.